python/cudf/cudf/_cuda/gpu.pyx

```diff
@@ -20,6 +20,28 @@ from enum import IntEnum
 from cudf._cuda.gpu cimport underlying_type_attribute as c_attr
 
 
+notify_caller_errors = {
+    cudaErrorInitializationError,
+    cudaErrorInsufficientDriver,
+    cudaErrorInvalidDeviceFunction,
+    cudaErrorNoDevice,
+    cudaErrorInvalidDevice,
+    cudaErrorStartupFailure,
+    cudaErrorInvalidKernelImage,
+    cudaErrorDeviceUninitialized,
+    cudaErrorAlreadyAcquired,
+    cudaErrorOperatingSystem,
+    cudaErrorNotPermitted,
+    cudaErrorNotSupported,
+    cudaErrorSystemNotReady,
+    cudaErrorSystemDriverMismatch,
+    cudaErrorCompatNotSupportedOnDevice,
+    cudaErrorTimeout,
+    cudaErrorUnknown,
+    cudaErrorApiFailureBase
+}
```

Comment:
Can these be defined in the `gpu_utils` file possibly? These are an implementation detail of our utility, not the cuda module.

Reply:
Moved this to `gpu_utils` by introducing `CudaError` class to expose the underlying `cudaError_t` enum